### PR TITLE
Remove txpool access

### PR DESCRIPTION
**Pull request by vimukthi-git**
<This is intended as a guide when describing-reviewing pull requests for go-centrifuge>
Issue Link: #715

### CODE REVIEW

File: ethereum/mock_ethclient.go [hunk 1]
```diff
@@ -1,16 +1,14 @@
 // +build integration unit
 
-package testingcommons
```
Comment (vimukthi-git) [inline ethereum/mock_ethclient.go]: had to change this because of dep cycles.

File: ethereum/geth_client.go [hunk 2]
```diff
@@ -5,12 +5,12 @@ import (
...
 	"net/url"
 	"reflect"
-	"sort"
-	"strconv"
 	"strings"
 	"sync"
 	"time"
 
+	"github.com/ethereum/go-ethereum"
+
```
Comment (vedhavyas) [inline ethereum/geth_client.go]: import spacing

**Comment by mikiquantum** [comment]
@vedhavyas In systems like infura you will never have certainty of the transaction queue nor pending transactions (unless they build up a common pool per API account). This fix focuses on not relaying on the internal txpool as it is not exposed in these type of clients. We can talk later about how parity does it, but it needs to be compatible with all clients. If this is still not clear let's sync offline.

File: ethereum/geth_client.go [hunk 3]
```diff
@@ -373,59 +382,17 @@ type callContexter interface {
...
-// else we increment the greater of current nonce or the nonce derived from txpool
-func (gc *gethClient) incrementNonce(opts *bind.TransactOpts, txpoolAccessEnabled bool, noncer noncer, cc callContexter) error {
-	// check if the txpool access is enabled
-	if !txpoolAccessEnabled {
-		log.Warningf("Ethereum Client doesn't support txpool API, may cause transactions failures")
-		opts.Nonce = nil
-		return nil
-	}
-
+func (gc *gethClient) incrementNonce(opts *bind.TransactOpts) error {
```
Comment (mikiquantum) [inline ethereum/geth_client.go]: I would rename it to:
```suggestion
func (gc *gethClient) setNonce(opts *bind.TransactOpts) error {
```

**Comment by vimukthi-git** [comment]
@mikiquantum as agreed online, anchoring concurrency test is a good test of this and we need to use contract methods in eth package to add a test like you mention. Therefore I will not add a test here.

**Pull request merged**